plugins/commands.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+const joi = require('joi');
+const boom = require('boom');
+const requestApi = require('request');
+const SCHEMA_COMMAND_NAMESPACE = joi.string().label('Command namespace');
+const SCHEMA_COMMAND_NAME = joi.string().label('Command name');
+const SCHEMA_COMMAND_VERSION = joi.string().label('Command version');
+const DEFAULT_TTL = 21 * 24 * 60 * 60 * 1000; // 3 weeks
+const DEFAULT_BYTES = 1024 * 1024 * 1024; // 1GB
+
+/**
+ * Determine if a pipeline can publish the command
+ * @method canPublishCommand
+ * @param  {String}  namespace  Command namespace
+ * @param  {String}  name       Command name
+ * @param  {Integer} pipelineId pipelineId
+ * @param  {String}  baseurl    API URL
+ * @return {Promise}
+ */
+function canPublishCommand(namespace, name, pipelineId, baseurl) {
+    return new Promise((resolve, reject) => {
+        requestApi({
```

Comment:
probably want to add store as allowCors to make it work: https://github.com/screwdriver-cd/screwdriver/blob/master/lib/server.js#L67

Reply:
Add store to corsOrigin. https://github.com/screwdriver-cd/screwdriver/pull/888

Reply:
Adding store to CORS is not necessary because the command access to the API is not via the browser. https://github.com/screwdriver-cd/screwdriver/pull/888#issuecomment-366155863 So probably it will work without adding store to CORS.